client/src/dependencies/DependencyManager.ts

```diff
@@ -130,25 +124,72 @@ export class DependencyManager {
                 const shell: boolean = (process.platform === 'win32') ? true : false;
 
                 try {
-                    const architecture: string = process.arch; // Returns the architecture Code is running on
-                    await CommandUtil.sendCommandWithOutput('npm', ['rebuild', dependency.moduleName, '--target=4.1.5', '--runtime=electron', '--dist-url=https://atom.io/download/electron', '--update-binary', '--fallback-to-build', `--target_arch=${architecture}`], extensionPath, null, outputAdapter, shell);
+                    await CommandUtil.sendCommandWithOutput('npm', ['rebuild', dependency, `--target=${info.longVersion}`, '--runtime=electron', '--dist-url=https://atom.io/download/electron', '--update-binary', '--fallback-to-build', `--target_arch=${architecture}`], extensionPath, null, outputAdapter, shell);
 
                 } catch (error) {
                     outputAdapter.log(LogType.ERROR, `Could not rebuild native dependencies ${error.message}. Please ensure that you have node and npm installed`);
                     throw error;
                 }
 
-                progress.report({message: `Updating ${dependency.moduleName}`});
-                outputAdapter.log(LogType.INFO, undefined, `Updating ${dependency.moduleName}`);
-                await fs.remove(`${extensionPath}/${dependency.original}`);
-                await fs.rename(`${extensionPath}/${dependency.new}`, `${extensionPath}/${dependency.original}`);
+                progress.report({message: `Updating ${dependency}`});
+                outputAdapter.log(LogType.INFO, undefined, `Updating ${dependency}`);
+                const basePath: string = path.join(extensionPath, 'node_modules', 'grpc', 'src', 'node', 'extension_binary');
+
+                const origPath: string = path.join(basePath, `node-v${info.modules}-${os}-${architecture}-${thing}`);
+                const newPath: string = path.join(basePath, `electron-v${info.shortVersion}-${os}-${architecture}-${thing}`);
+
+                const exists: boolean = await fs.pathExists(origPath);
+                if (exists) {
+                    await fs.remove(origPath);
+                }
+                await fs.rename(newPath, origPath);
             }
 
             outputAdapter.log(LogType.SUCCESS, undefined, 'Finished updating native node modules');
             progress.report({message: 'Finished updating native node modules'});
         });
     }
 
+    private async getRebuildInfo(os: string, arch: string, thing: string): Promise<{ modules: string, longVersion: string, shortVersion: string }> {
+        const response: any = await Axios.get('https://raw.githubusercontent.com/electron/releases/master/lite.json');
```

Comment:
Should we be try/catching this, or even setting a timeout for how long this should take? What happens if the user doesn't have an internet connection?

Reply:
If the user doesn't have an internet connection then none of the getting the grpc prebuilt binary code will work. The error will eventually be caught by failing to activate.